### PR TITLE
Workaround for crash on Samsung devices on Android 4.2

### PR DESCRIPTION
**Pull request by timrae**
https://code.google.com/p/android/issues/detail?id=78377

### CODE REVIEW

**Comment by hssm** [comment]
I'd say this is serious enough to warrant a release really soon.

**Comment by timrae** [comment]
I agree, is there anything else you want to quickly put in 2.5.2?

File: AnkiDroid/proguard-rules.pro [hunk 1]
```diff
@@ -15,3 +15,13 @@
...
+
+# Samsung Android 4.2 bug workaround
+# https://code.google.com/p/android/issues/detail?id=78377
+-keepattributes **
+-keep class !android.support.v7.view.menu.**,!android.support.design.internal.NavigationMenu,!android.support.design.internal.NavigationMenuPresenter,!android.support.design.internal.NavigationSubMenu,** {*;}
+-dontpreverify
+-dontoptimize
+-dontshrink
+-dontwarn **
+-dontnote **
```
Comment (hssm) [inline AnkiDroid/proguard-rules.pro]: I don't really know the proguard syntax but I imagine this is turning off almost all of its features?  I know we have some dynamically invoked functions in libanki that might get stripped out by proguard if we tell it not to.

**Comment by timrae** [comment]
The Android 6 fix is also relatively important

**Comment by hssm** [comment]
No, I have nothing else to add today.

File: AnkiDroid/proguard-rules.pro (hunk 1 above)
Comment (timrae) [inline AnkiDroid/proguard-rules.pro]: Yeah neither do I (I just copy pasted tbh lol), but as far as I understand this obfuscates the class names mentioned, and leaves everything else untouched.

Comment (timrae) [inline AnkiDroid/proguard-rules.pro]: Please give it a good testing and I'll release 2.5.2 when I get up in the morning

Comment (timrae) [inline AnkiDroid/proguard-rules.pro]: I checked the [mapping.txt](https://github.com/ankidroid/Anki-Android/files/51190/mapping.txt) file and the contents of that seem to agree with the operation that I described above:

The majority of the members map to themselves, but for the classes explicitly mentioned in the config we see mappings like

`android.support.v7.view.menu.MenuPresenter$Callback mCallback -> e`
`android.support.design.internal.NavigationMenu -> android.support.design.internal.a`

Comment (hssm) [inline AnkiDroid/proguard-rules.pro]: Makes sense. Thanks for looking into it.

**Pull request merged**